src/components/HourlyForecast.vue

```diff
@@ -0,0 +1,292 @@
+<template functional>
+  <div class="hourly-forecast">
+    <table class="hourly-forecast__data-table data-table">
+      <tbody>
+        <tr>
+          <th class="data-table__headings-column">День тижня</th>
+          <th
+            v-for="(day, index) in props.weather.datesWithColumnsNumber"
+            :key="index"
+            class="data-table__day-row"
+            :colspan="day.columnsNumber"
+          >
+            <div class="data-table__day" :data-tooltip="day.tooltipString">
+              {{ day.displayString }}
+            </div>
+          </th>
+        </tr>
+        <tr>
+          <th class="data-table__headings-column">Час</th>
+          <th
+            v-for="(hourForecast, index) in props.weather.data"
+            :key="index"
+            class="data-table__time cell"
+            v-bind:class="{
+              'cell_left-border': hourForecast.beginNextDay,
+              cell_shadow: !hourForecast.isDayTime,
+            }"
+          >
+            {{ hourForecast.time }}
+          </th>
+        </tr>
+        <tr>
+          <td class="data-table__headings-column">Опис погоди</td>
+          <td
+            v-for="(hourForecast, index) in props.weather.data"
+            :key="index"
+            class="data-table__icon icon cell"
+            :data-tooltip="hourForecast.weatherDescription"
+            v-bind:class="{
+              'cell_left-border': hourForecast.beginNextDay,
+              cell_shadow: !hourForecast.isDayTime,
+            }"
+          >
+            <img :src="hourForecast.icon" alt="weather icon" class="data-table__icon-image" />
+          </td>
+        </tr>
+        <tr>
+          <td class="data-table__headings-column">Температура, °C</td>
+          <td
+            v-for="(hourForecast, index) in props.weather.data"
+            :key="index"
+            class="data-table__temperature cell"
+            v-bind:class="{
+              'cell_left-border': hourForecast.beginNextDay,
+              cell_shadow: !hourForecast.isDayTime,
```

Comment:
Are you using BEM? Because by BEM this class should be `cell--with-shadow`

Reply:
@probil, do you mean `cell_with-shadow`? Look at [there](https://ru.bem.info/methodology/quick-start/#%D0%9C%D0%BE%D0%B4%D0%B8%D1%84%D0%B8%D0%BA%D0%B0%D1%82%D0%BE%D1%80)

Reply:
I meant previous edition of BEM.
In any case my comment was optional.